depmapomics/upload.py

```diff
@@ -0,0 +1,1298 @@
+            {
+                "path": folder
+                + "all_somatic_mutations_boolmatrix_fordepmap_damaging.csv",
+                "name": "CCLE_mutations_bool_damaging",
+                "format": "NumericMatrixCSV",
+                "encoding": "utf-8",
+            },
+            {
+                "path": folder
+                + "all_somatic_mutations_boolmatrix_fordepmap_othernoncons.csv",
+                "name": "CCLE_mutations_bool_nonconserving",
+                "format": "NumericMatrixCSV",
+                "encoding": "utf-8",
+            },
+            # {
+            #     "path": folder
+            #     + "all_somatic_mutations_boolmatrix_fordepmap_othercons.csv",
+            #     "name": "CCLE_mutations_bool_otherconserving",
+            #     "format": "NumericMatrixCSV",
+            #     "encoding": "utf-8",
+            # },
+            {
+                "path": folder
+                + "all_somatic_mutations_boolmatrix_fordepmap_hotspot.csv",
+                "name": "CCLE_mutations_bool_hotspot",
+                "format": "NumericMatrixCSV",
+                "encoding": "utf-8",
+            },
+        ],
+        add_all_existing_files=True,
+    )
+
+
+def uploadMutationMatricesModel(
+    pr2model_dict, portal, taiga_latest=TAIGA_MUTATION, taiga_virtual=""
+):
+    """subset, rename, save and upload to taiga model-level mutation matrices
+    
+    Args:
+        pr2model_dict (dict): renaming scheme mapping from PR-id to model id
+        portal (str): which portal the data is being uploaded to
+        taiga_latest (str): which dataset the matrices to be subsetted are being read from
+        taiga_virtual (str): which dataset the matrices are being uploaded to
+    """
+    folder = "temp/" + portal + "/model/"
+    h.createFoldersFor(folder)
+    # load cds-id indexed matrices for the current quarter
+    print("Mutation: loading from taiga latest")
+    tc = TaigaClient()
+    mutations = tc.get(
+        name=taiga_latest, file="merged_somatic_mutations_fordepmap_profile"
+    )
+    damaging = tc.get(
+        name=taiga_latest, file="merged_somatic_mutations_boolmatrix_fordepmap_damaging"
+    )
+    # othercons = tc.get(
+    #     name=taiga_latest,
+    #     file="merged_somatic_mutations_boolmatrix_fordepmap_othercons",
+    # )
+    othernoncons = tc.get(
+        name=taiga_latest,
+        file="merged_somatic_mutations_boolmatrix_fordepmap_othernoncons",
+    )
+    hotspot = tc.get(
+        name=taiga_latest, file="merged_somatic_mutations_boolmatrix_fordepmap_hotspot"
+    )
+
+    # subset and rename
+    print("Mutation: subsetting and renaming")
+    mutations_renamed = mutations[
+        mutations.DepMap_ID.isin(set(pr2model_dict.keys()))
+    ].replace({"DepMap_ID": pr2model_dict})
+    mutations_renamed.to_csv(folder + "all_somatic_mutations_model.csv", index=False)
+    damaging_renamed = damaging[damaging.index.isin(set(pr2model_dict.keys()))].rename(
```

Comment:
this block looks like it's repeated. In the vein of "self documenting code" could you encapsulate this into a function which has a descriptive name and you call it once per matrix? I _think_ these are all the same operation, but it'd be clearer if they were calling an identical function.

Reply:
@5im1z